netbox/utilities/views.py

```diff
@@ -93,7 +94,7 @@ def dispatch(self, request, *args, **kwargs):
                 'a base queryset'.format(self.__class__.__name__)
             )
 
-        if not self.has_permission():
+        if isinstance(self.queryset, RestrictedQuerySet) and not self.has_permission():
```

Comment:
We don't want to bypass `has_permission()` entirely because it's responsible for calling `user.has_perms()`. If it's necessary to avoid calling `restrict()` on querysets which don't support it, we should account for that inside `has_permission()`.